scripts/menu/server/sv_menu.lua

```diff
@@ -0,0 +1,405 @@
+    if resultHeaders['Set-Cookie'] then
+      local cookieHeader = resultHeaders['Set-Cookie']
+      local cookies = type(cookieHeader) == 'table' and cookieHeader or { cookieHeader }
+      
+      for k in pairs(cookies) do
+        cookies[k] = cookies[k] .. '; SameSite=None; Secure'
+      end
+      
+      resultHeaders['Set-Cookie'] = cookies
+    end
+
+    -- Sniff permissions out of the auth request
+    if path == '/auth/nui' and httpCode == 200 then
+      local resp = json.decode(data)
+      if resp and resp.isAdmin then
+        debugPrint("Caching admin " .. s .. " permissions: " .. json.encode(resp.permissions))
+        adminPermissions[s] = resp.permissions
+      else
+        adminPermissions[s] = nil
+      end
+    end
+  
+    local errorCode = tonumber(httpCode) >= 400
+    local resultColor = errorCode and '^1' or '^2'
+    debugPrint(string.format(
+      "^3WebPipe[^5%d^0:^1%d^3]^0 %s<< %s ^4%s^0", s, callbackId, resultColor, httpCode, path))
+    if errorCode then
+      debugPrint(string.format(
+        "^3WebPipe[^5%d^0:^1%d^3]^0 %s<< Headers: %s^0", s, callbackId, resultColor, json.encode(resultHeaders)))
+    end
+    
+    TriggerClientEvent('txAdmin:WebPipe', s, callbackId, httpCode, data, resultHeaders)
+  end, method, body, headers, {
+    followLocation = false
+  })
+end)
+--
+-- [[ End WebPipe ]]
+--
+
+--
+-- [[ ServerCtxObj ]]
+--
+local ServerCtxObj = {
+  oneSync = {
+    type = nil,
+    status = false
+  },
+  projectName = nil,
+  maxClients = 30,
+  locale = nil
+}
+
+local function syncServerCtx()
+  local oneSyncConvar = GetConvar('onesync', 'off')
+  if oneSyncConvar == ('on' or 'legacy') then
+    ServerCtxObj.oneSync.type = oneSyncConvar
+    ServerCtxObj.status = true
+  elseif oneSyncConvar == 'off' then
+    ServerCtxObj.oneSyncStatus = false
+  end
+
+  -- Default '' in fxServer
+  local svProjectName = GetConvar('sv_projectname', '')
+  if svProjectName ~= '' then
+    ServerCtxObj.projectName = svProjectName
+  end
+
+  -- Default 30 in fxServer
+  local svMaxClients = GetConvarInt('sv_maxclients', 30)
+  ServerCtxObj.maxClients = svMaxClients
+
+  local txAdminLocale = GetConvar('txAdmin-locale', 'en')
+  ServerCtxObj.locale = txAdminLocale
+
+  debugPrint('Server CTX assigned to GlobalState, CTX:')
+  debugPrint(json.encode(ServerCtxObj))
+  GlobalState.txAdminServerCtx = ServerCtxObj
+end
+
+-- Everytime the txAdmin convars are changed this event will fire
+-- Therefore, lets update global state with that.
+AddEventHandler('txAdmin:events:configChanged', function()
+  debugPrint('configChanged event triggered, syncing GlobalState')
+  syncServerCtx()
+end)
+
+CreateThread(function()
+  -- If we don't wait for a tick there is some race condition that
+  -- sometimes prevents debugPrint lmao
+  Wait(0)
+  syncServerCtx()
+end)
+
+RegisterServerEvent('txAdmin:menu:getServerCtx', function()
+  local src = source
+  TriggerClientEvent('txAdmin:menu:sendServerCtx', src, serverCtxObj)
```

Comment:
Is this still used? Seems like a great way to get oneself spammed with events to reject and clog up the server send list otherwise.